met-web/src/components/engagement/listing/index.tsx

```diff
@@ -252,10 +238,57 @@ const EngagementListing = () => {
-                            {advancedSearchOpen ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />} Advanced Search
-                        </SecondaryButton>
+                        <When condition={!isMediumScreen}>
+                            <SecondaryButton
+                                data-testid="engagement/listing/advancedSearch"
+                                name="advancedSearch"
+                                onClick={() => setAdvancedSearchOpen(!advancedSearchOpen)}
+                            >
+                                {
+                                    <ExpandMoreIcon
+                                        sx={[
+                                            {
+                                                transform: 'rotate(0deg)',
+                                                transition: (theme) =>
+                                                    theme.transitions.create('all', {
+                                                        duration: theme.transitions.duration.shortest,
+                                                    }),
+                                            },
+                                            advancedSearchOpen && {
+                                                transform: 'rotate(180deg)',
+                                            },
+                                        ]}
+                                    />
+                                }
+                                {}Advanced Search
+                            </SecondaryButton>
+                        </When>
                     </Stack>
+                    <When condition={isMediumScreen}>
+                        <SecondaryButton
+                            data-testid="engagement/listing/advancedSearch"
+                            name="advancedSearch"
+                            onClick={() => setAdvancedSearchOpen(!advancedSearchOpen)}
+                        >
+                            {
+                                <ExpandMoreIcon
+                                    sx={[
+                                        {
+                                            transform: 'rotate(0deg)',
+                                            transition: (theme) =>
+                                                theme.transitions.create('all', {
+                                                    duration: theme.transitions.duration.shortest,
+                                                }),
+                                        },
+                                        advancedSearchOpen && {
+                                            transform: 'rotate(180deg)',
+                                        },
+                                    ]}
+                                />
+                            }
+                            {}Advanced Search
+                        </SecondaryButton>
+                    </When>
```

Comment:
what's the difference between medium screen button and large screen button?

Reply:
the difference is just the position of the button.

Reply:
We need to improve on this, there's too much duplication. We should find a way to leverage Grid to place it one code for both views